DANS-KNAW/easy-deposit-api
# EASY-1820 Change acceptLicenseAgreement to acceptDepositAgreement

Fixes EASY-1820 Change acceptLicenseAgreement to acceptDepositAgreement 

#### When applied it will
* have licenseAgreement renamed to depositAgreement in scala objects that are (de)serialized to json
* mention explicit versions of XSD so nothing will break when EASY-SCHEMA upgrades any XSD
* validate the XML files with schema's for files.xml and agreements.xml

#### Where should the reviewer @DANS-KNAW/easy start?

#### How should this be manually tested?

* build and deploy together with ui
  * temporarily increment https://github.com/DANS-KNAW/easy-deposit-api/blob/8d3fa55954a21a7f29fcc170959413fea5f718b5/pom.xml#L28
  * as long as tests fail: `mvn clean install -DskipTests=true`
* check that the service runs: `sudo service easy-deposit-api status`
  or `curl -i 'http://deasy.dans.knaw.nl:20190/'`
* go to `http://deasy.dans.knaw.nl/deposit` with a browser, login, create a deposit
* check the box at the bottom of the form to accept the depositor agreement
* [x] save a draft
* submit and examine on deasy:

      /var/opt/dans.knaw.nl/tmp/easy-ingest-flow-inbox/dda8035e-ac46-4c74-ba08-81a4ebb7d8f7/bag/metadata/agreements.xml

#### Related pull requests on github
* revert #89 before merging this one, it requires red-hat 7
* must be merged/deployed together with https://github.com/DANS-KNAW/easy-deposit-ui/pull/113
* Once these two are _merged and deployed_:
  * first apply/merge/publish rename in [agreements.xsd](https://github.com/DANS-KNAW/easy-schema/blob/731999908d836cd2f7bd66d4e7e782fadfe0c0f6/src/main/assembly/dist/bag/metadata/agreements/2018/05/agreements.xsd#L45)  of easy-schema
  * then update version number in
    * https://github.com/DANS-KNAW/easy-deposit-api/blob/cbd6e2c8c23a55f1dfe2e646f362d853fa103f6c/src/main/scala/nl.knaw.dans.easy.deposit/docs/AgreementsXml.scala#L25
    * [`application.properties`](https://github.com/DANS-KNAW/easy-validate-dans-bag/blob/076bd82cd95d365d2826bf932ba4e946f9cb5b0c/src/main/assembly/dist/cfg/application.properties#L11) of `easy-validate-dans-bag`
